src/test/java/bio/terra/integration/DataRepoFixtures.java

```diff
@@ -687,6 +691,50 @@ public SnapshotModel getSnapshot(
     return validateResponse(response, "snapshot retrieve", HttpStatus.OK, null);
   }
 
+  /**
+   * Given a dataset, table, and column, retrieve the DRS URI from the snapshot preview and extract
+   * its DRS object ID.
+   *
+   * <p>In the past, we've queried BigQuery directly for this, but unpredictable delays in IAM
+   * propagation make going through the preview more reliable (TDR proxies its request to BigQuery
+   * here).
+   */
+  public String retrieveDrsIdFromSnapshotPreview(
+      TestConfiguration.User user, UUID snapshotId, String table, String column) throws Exception {
+    String filter = "WHERE %s IS NOT NULL".formatted(column);
+    DataRepoResponse<SnapshotPreviewModel> response =
+        retrieveSnapshotPreviewByIdRaw(user, snapshotId, table, 0, 1, filter);
+    SnapshotPreviewModel validated =
+        validateResponse(response, "snapshot preview for DRS ID", HttpStatus.OK, null);
+
+    assertThat("Got one row", validated.getResult(), hasSize(1));
```

Comment:
Do we want to enforce this or just return the first one we find?

Reply:
So I was trying to port over the assertions and general functionality from the old BQ fixture, and this was something we asserted.  I think it's reasonable to include in this new version: it would be unexpected to have an empty result set at this stage, and we've already limited the results to 1.